src/endpoint/s3/s3_bucket_logging.js

```diff
@@ -54,15 +56,30 @@ const create_syslog_udp_socket = (() => {
 })();
 
 
-function endpoint_bucket_op_logs(op_name, req, res, source_bucket) {
+async function endpoint_bucket_op_logs(op_name, req, res, source_bucket) {
 
     // 1 - Get all the information to be logged in a log message.
     // 2 - Format it and send it to log bucket/syslog.
     const s3_log = get_bucket_log_record(op_name, source_bucket, req, res);
     dbg.log1("Bucket operation logs = ", s3_log);
 
+    switch (config.BUCKET_LOG_TYPE) {
```

Comment:
It seems that this function will be invoked for each request. Can we create a persistent logger in a higher scope so that we don't have to `open` and `close` the log file every time?

The log file seems to be getting processed using `.process` which should take care of replacing the old file as well as the logger should be able to detect the replacement and give up the locks.

Reply:
Will take a look on that

Reply:
done! moved up to req.bucket_logger.